executor/pkg_test.go

```diff
@@ -140,3 +242,33 @@ func (s *pkgTestSuite) TestMoveInfoSchemaToFront(c *C) {
 		}
 	}
 }
+
+func BenchmarkPartitionInnerRows(b *testing.B) {
```

Comment:
The benchmark result of partition 1k rows:
```
➜ go test -bench BenchmarkPartitionInnerRows -run xx -count 5 --benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/executor
BenchmarkPartitionInnerRows-8               3000            543577 ns/op          208878 B/op       3505 allocs/op
BenchmarkPartitionInnerRows-8               3000            495717 ns/op          208876 B/op       3505 allocs/op
BenchmarkPartitionInnerRows-8               3000            572750 ns/op          208873 B/op       3505 allocs/op
BenchmarkPartitionInnerRows-8               2000            526905 ns/op          208660 B/op       3505 allocs/op
BenchmarkPartitionInnerRows-8               3000            547324 ns/op          208871 B/op       3505 allocs/op
PASS
ok      github.com/pingcap/tidb/executor        8.046s
```

10K rows:
```
➜ go test -bench BenchmarkPartitionInnerRows -run xx -count 5 --benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/executor
BenchmarkPartitionInnerRows-8                300           5513236 ns/op         2077763 B/op      35006 allocs/op
BenchmarkPartitionInnerRows-8                300           4567255 ns/op         2077784 B/op      35006 allocs/op
BenchmarkPartitionInnerRows-8                500           6000812 ns/op         2070703 B/op      35005 allocs/op
BenchmarkPartitionInnerRows-8                300           5584459 ns/op         2077753 B/op      35006 allocs/op
BenchmarkPartitionInnerRows-8                200           5567495 ns/op         2085347 B/op      35006 allocs/op
PASS
ok      github.com/pingcap/tidb/executor        13.006s
```

1M rows:
```
➜ go test -bench BenchmarkPartitionInnerRows -run xx -count 5 --benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/executor
BenchmarkPartitionInnerRows-8                  2         641133830 ns/op        208521144 B/op   3500195 allocs/op
BenchmarkPartitionInnerRows-8                  3         632080200 ns/op        207761786 B/op   3500135 allocs/op
BenchmarkPartitionInnerRows-8                  3         632613116 ns/op        207761408 B/op   3500134 allocs/op
BenchmarkPartitionInnerRows-8                  3         396137105 ns/op        207761648 B/op   3500134 allocs/op
BenchmarkPartitionInnerRows-8                  3         629841188 ns/op        207761280 B/op   3500133 allocs/op
PASS
ok      github.com/pingcap/tidb/executor        31.418s
```

Reply:
The cache size, chunk size, part concurrency is not correct.

I re-run the bench test.

100w rows:
```
➜ go test -bench BenchmarkPartitionInnerRows -run xx -count 5 --benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/executor

BenchmarkPartitionInnerRows-4                 5  257704061 ns/op
BenchmarkPartitionInnerRows-4                 5  250817868 ns/op
BenchmarkPartitionInnerRows-4                 5  274229203 ns/op
BenchmarkPartitionInnerRows-4                 5  263210431 ns/op
BenchmarkPartitionInnerRows-4                 3  336262754 ns/op
```

Reply:
I tested a bench for join.

part:     200w rows
test_xhy: 5 rows
``` sql
mysql> select distinct p_mfgr from part;
+----------------+
| p_mfgr         |
+----------------+
| Manufacturer#1 |
| Manufacturer#2 |
| Manufacturer#3 |
| Manufacturer#4 |
| Manufacturer#5 |
+----------------+
5 rows in set (2.06 sec)

mysql> select * from test_xhy;
+----------------+
| a              |
+----------------+
| Manufacturer#1 |
| Manufacturer#2 |
| Manufacturer#3 |
| Manufacturer#4 |
| Manufacturer#5 |
+----------------+
5 rows in set (0.01 sec)

mysql> select /*+ TIDB_HJ(t1,t2) */ count(*) from test_xhy t1 left join part t2 on t1.a=t2.p_mfgr;
+----------+
| count(*) |
+----------+
|  2000000 |
+----------+
1 row in set (2.43 sec)

mysql> explain analyze select /*+ TIDB_HJ(t1,t2) */ count(*) from test_xhy t1 left join part t2 on t1.a=t2.p_mfgr;
+--------------------------+------------+------+--------------------------------------------------------------------+---------------------------------------------+
| id                       | count      | task | operator info                                                      | execution info                              |
+--------------------------+------------+------+--------------------------------------------------------------------+---------------------------------------------+
| StreamAgg_11             | 1.00       | root | funcs:count(1)                                                     | time:2.456780995s, loops:2, rows:1          |
| └─HashLeftJoin_17        | 2000000.00 | root | left outer join, inner:TableReader_16, equal:[eq(t1.a, t2.p_mfgr)] | time:2.381185288s, loops:1955, rows:2000000 |
|   ├─TableReader_19       | 5.00       | root | data:TableScan_18                                                  | time:7.595562ms, loops:2, rows:5            |
|   │ └─TableScan_18       | 5.00       | cop  | table:t1, range:[-inf,+inf], keep order:false, stats:pseudo        |                                             |
|   └─TableReader_16       | 2000000.00 | root | data:TableScan_15                                                  | time:1.249685953s, loops:1955, rows:2000000 |
|     └─TableScan_15       | 2000000.00 | cop  | table:t2, range:[-inf,+inf], keep order:false                      |                                             |
+--------------------------+------------+------+--------------------------------------------------------------------+---------------------------------------------+
6 rows in set (2.46 sec)
```